logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/SearchUserRepository.kt

```diff
@@ -64,8 +80,32 @@ class SearchUserRepositoryImpl(
         wrapApiRequest {
             userDetailsApi.getMultipleUsers(ListUserRequest.qualifiedIds(contactResultValue.documents.map { it.qualifiedID }))
         }.map { userDetailsResponses ->
-            UserSearchResult(publicUserMapper.fromUserDetailResponses(userDetailsResponses))
+            UserSearchResult(userDetailsResponses.map { userProfileDTO ->
+                publicUserMapper.fromUserDetailResponseWithUsertype(
+                    userDetailResponse = userProfileDTO,
+                    userType = userTypeMapper.fromOtherUserTeamAndDomain(
+                        otherUserDomain = userProfileDTO.id.domain,
+                        selfUserTeamId = getSelfUser().teamId,
```

Comment:
We can pass the `selfUser.teamId` as a parameter to this function, and the `UseCase` can figure it out by using the `UserRepository`.

Reply:
will add a TODO